cfgmgr/buffer_check_headroom_mellanox.lua

```diff
@@ -55,17 +59,31 @@ end
 
 local asic_keys = redis.call('KEYS', 'ASIC_TABLE*')
 local pipeline_latency = tonumber(redis.call('HGET', asic_keys[1], 'pipeline_latency'))
+local cell_size = tonumber(redis.call('HGET', asic_keys[1], 'cell_size'))
+local port_reserved_shp = tonumber(redis.call('HGET', asic_keys[1], 'port_reserved_shp'))
+local port_max_shp = tonumber(redis.call('HGET', asic_keys[1], 'port_max_shp'))
 if is_port_with_8lanes(lanes) then
     -- The pipeline latency should be adjusted accordingly for ports with 2 buffer units
     pipeline_latency = pipeline_latency * 2 - 1
     egress_mirror_size = egress_mirror_size * 2
+    port_reserved_shp = port_reserved_shp * 2
 end
+
 local lossy_pg_size = pipeline_latency * 1024
 accumulative_size = accumulative_size + lossy_pg_size + egress_mirror_size
 
 -- Fetch all keys in BUFFER_PG according to the port
 redis.call('SELECT', appl_db)
 
+local is_shp_enabled
+local shp_size = tonumber(redis.call('HGET', 'BUFFER_POOL_TABLE:ingress_lossless_pool', 'xoff'))
+if shp_size == nil or shp_size == 0 then
```

Comment:
Can we check shared headroom pool based on the mode dynamic or static??

Reply:
No need to check that. All the Lua plugins are invoked only in dynamic mode